### PR TITLE
monitoring:package SNMP MIB file as an rpm

### PR DESCRIPTION
**Pull request by bigjust**
Adds a new subpackage 'ceph-mib' that installs the Ceph MIB file in a
standard location '/usr/share/snmp/CEPH-MIB.txt'. This allows users to
use their own snmp manager in order to receive events from Ceph

Fixes: https://tracker.ceph.com/issues/56433
Signed-off-by: Justin Caratzas jcaratza@redhat.com

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -2562,4 +2572,8 @@ exit 0
 %attr(0755,root,root) %dir %{_sysconfdir}/prometheus/ceph
 %config %{_sysconfdir}/prometheus/ceph/ceph_default_alerts.yml
 
+%files mib
+%attr(0755,root,root) %dir %{_datadir}/snmp
+%attr(0655,-,-,) %{_datadir}/snmp/mib
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply ceph.spec.in:2577]: i am curious why do we need to specify the access bits explicitly? is 
```
%dir %{_datadir}/snmp/mibs
```
good enough?

File: ceph.spec.in [hunk 2]
```diff
@@ -1244,6 +1244,13 @@ Group:          System/Monitoring
 %description prometheus-alerts
 This package provides Ceph default alerts for Prometheus.
 
+%package mib
+Summary:        MIB for SNMP alerts
+BuildArch:      noarch
+Group:          System/Monitoring
```
Comment (tchaikov) [inline ceph.spec.in:1250]: might want to guard this with `%if 0%{?suse_version}`?

Comment (tchaikov) [reply ceph.spec.in:1251]: @bigjust hi Justin, sorry for the confusion, i meant to suggest that we should guard
```
Group: System/Monitoring
```
with  `%if 0%{?suse_version}` as the `Group` is a SUSE only thing. see https://en.opensuse.org/openSUSE:Package_group_guidelines . while fedora stopped using it, see https://fedoraproject.org/wiki/RPMGroups

Comment (bigjust) [reply ceph.spec.in:1251]: Ah I see! Thanks for the guidance :), I've updated the commit

**Review by tchaikov — APPROVED** [review]

**Comment by bigjust** [comment]
jenkins test api

**Comment by djgalloway** [comment]
This (I think?) is breaking CentOS builds.

```
Processing files: ceph-mib-17.0.0-13555.g795daff9.el8.noarch
error: Directory not found: /home/jenkins-build/build/workspace/ceph-dev-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-13555-g795daff9/rpm/el8/BUILDROOT/ceph-17.0.0-13555.g795daff9.el8.x86_64/usr/share/snmp
error: Directory not found: /home/jenkins-build/build/workspace/ceph-dev-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-13555-g795daff9/rpm/el8/BUILDROOT/ceph-17.0.0-13555.g795daff9.el8.x86_64/usr/share/snmp/mibs


RPM build errors:
    Ignoring invalid regex ^%{_scl_prefix}/.*|%{_root_sysconfdir}/rpm/macros.gcc-toolset-11-config$
    Ignoring invalid regex ^%{_root_sysconfdir}/rpm/macros.gcc-toolset-11-config$
    Ignoring invalid regex ^%{_scl_prefix}/.*|%{_root_sysconfdir}/rpm/macros.gcc-toolset-11-config$
...
```

https://jenkins.ceph.com/job/ceph-dev-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/49994/consoleFull

**Comment by dmick** [comment]
I think it's because you need a dependency on net-snmp, so that /usr/share/snmp exists as a destination

**Comment by dmick** [comment]
no, that wasn't enough.  I suspect %files currently but I have to read up on specfiles

**Comment by bigjust** [comment]
It looks like a rpm macro substitution did not work, it created:

`/home/jenkins-build/build/workspace/ceph-dev-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-13555-g795daff9/rpm/el8/BUILDROOT/ceph-17.0.0-13555.g795daff9.el8.x86_64//snmp/mibs`

and then tried to install the file into:

`/home/jenkins-build/build/workspace/ceph-dev-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-13555-g795daff9/rpm/el8/BUILDROOT/ceph-17.0.0-13555.g795daff9.el8.x86_64/usr/share/snmp`

I'm testing a patch now

**Comment by idryomov** [comment]
@bigjust It looks like you merged this over Kefu's `needs-qa` label which would have included a build on all platforms at a minimum.  Please don't do that in the future.

**Comment by bigjust** [comment]
Understood, I'll submit a new PR

**Pull request merged**